msticpy/data/drivers/sumologic_driver.py

```diff
@@ -287,10 +291,12 @@ def _poll_job_status(self, searchjob, verbosity):
 
     # pylint: disable=inconsistent-return-statements
     # I don't think there are any - everything returns a list
-    def _get_job_results(self, searchjob, status, qry_count, force_mssg_rstls, limit):
+    def _get_job_results(
```

Comment:
This is a scarily long function. Is it possible to break it up into subfunctions that each handled the 4 cases:
- get_results_with_limit_force()
- get_results_no_limit_force()
- get_results_with_limit_gt_count()
- get_results_with_limit_lte_count()
- get_results_no_limit_gt_count()
- get_results_no_limit_lte_count()

Or something along those lines.
It would be easier to see the logic of how each option is selected and would abstract this from the mechanics of setting up the service query.

Reply:
done. that was already on my list. just ongoing work. split in 2 only is enough for linting to be happy.